include/zephyr/dsp/dsp.h

```diff
@@ -17,6 +17,18 @@
 #define DSP_FUNC_SCOPE
 #endif
 
+#ifdef CONFIG_DSP_BACKEND_HAS_AGU
+#define DSP_DATA __agu
+#else
+#define DSP_DATA
+#endif
+
+#ifdef CONFIG_DSP_BACKEND_HAS_XDATA_SECTION
+#define DATA_ATTR __attribute__((section(".Xdata")))
+#else
+#define DATA_ATTR
+#endif
```

Comment:
`DATA_ATTR` has multiple problems:

1. It is too generic -- if any, it should be called `DSP_DATA_ATTR`.
2. It is in conflict with `DSP_DATA` -- note that both `DSP_DATA` and `DATA_ATTR` are "data attributes."

In general, I would say `DATA_ATTR` is a misnomer and should be reworked as `#define DSP_STATIC_DATA DSP_DATA __attribute__((section(".Xdata")))`.